File: core/OddEvenLinkedList/OddEvenLinkedList.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def oddEvenList(head: Optional[ListNode]) -> Optional[ListNode]:
    if head is None:
        return head
    remain = head.next
    res, point = head, remain
    while point and point.next:
        res.next = point.next
        res = res.next
        point.next = res.next
        point = point.next
    res.next = remain
    return head

File: core/OddEvenLinkedList/OddEvenLinkedList.java
package com.dbc.code;

public class OddEvenLinkedList {
    public class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    public ListNode oddEvenList(ListNode head) {
        if (head == null) return head;
        ListNode remain = head.next;
        ListNode res = head, point = remain;
        while (point != null && point.next != null) {
            res.next = point.next;
            res = res.next;
            point.next = res.next;
            point = point.next;
        }
        res.next = remain;
        return head;
    }
}
